math/src/msm/pippenger.rs

```diff
@@ -86,10 +94,67 @@ where
                 .reduce(|g, m| g.operate_with(&m))
                 .unwrap_or_else(G::neutral_element)
         })
+        // NOTE: this operation is non-associative and strictly sequential
         .reduce(|t, g| t.operate_with_self(1_u64 << window_size).operate_with(&g))
         .unwrap_or_else(G::neutral_element)
 }
 
+#[cfg(feature = "rayon")]
+// It has the following differences with the sequential one:
+//  1. It uses one vec per thread to store buckets.
+//  2. It reduces all window results via a different method.
+pub fn parallel_msm_with<const NUM_LIMBS: usize, G>(
+    cs: &[UnsignedInteger<NUM_LIMBS>],
+    hidings: &[G],
```

Comment:
Could we name the second parameter `points` or something like that? In this abstract math context the term "hiding" doesn't really have a meaning.

Reply:
Sure. I was just following the convention on the original `msm` function. I'll change it in all of them then.

Reply:
Done